pandas/io/common.py

```diff
@@ -285,8 +285,45 @@ def ZipFile(*args, **kwargs):
     ZipFile = zipfile.ZipFile
 
 
-def _get_handle(path, mode, encoding=None, compression=None, memory_map=False):
+def _get_inferred_compression(filepath_or_buffer, compression):
+    if compression == 'infer':
+        if isinstance(filepath_or_buffer, compat.string_types):
+            if filepath_or_buffer.endswith('.gz'):
+                inferred_compression = 'gzip'
+            elif filepath_or_buffer.endswith('.bz2'):
+                inferred_compression = 'bz2'
+            elif filepath_or_buffer.endswith('.zip'):
+                inferred_compression = 'zip'
+            elif filepath_or_buffer.endswith('.xz'):
+                inferred_compression = 'xz'
+            else:
+                inferred_compression = None
+        else:
+            inferred_compression = None
+    else:
+        inferred_compression = compression
+    return inferred_compression
+
+
+def _get_handle(path, mode, encoding=None, compression=None, memory_map=False, is_txt=True):
     """Gets file handle for given path and mode.
+    
+    Parameters
+    ----------
+    path : string
+        file path
+    mode : string
+        mode to open file, like 'wb', 'rb', etc
+    encoding : string, default None
+        encoding for text file
+    compression : string, default None
```

Comment:
this should *also* take ``compression='infer'``, which if passed then infers the compression (IOW, call _get_infer_compression internally)